cpp/tests/utilities/column_wrapper.cuh

```diff
@@ -0,0 +1,396 @@
+#include <thrust/logical.h>
+#include <bitset>
+#include <string>
+
+#ifndef CUDA_RT_CALL
+#define CUDA_RT_CALL(call)                                                    \
+  do {                                                                        \
+    cudaError_t cudaStatus = (call);                                          \
+    if (cudaSuccess != cudaStatus) {                                          \
+      fprintf(stderr,                                                         \
+              "ERROR: CUDA RT call \"%s\" in line %d of file %s failed with " \
+              "%s (%d).\n",                                                   \
+              #call, __LINE__, __FILE__, cudaGetErrorString(cudaStatus),      \
+              cudaStatus);                                                    \
+      exit(1);                                                                \
+    }                                                                         \
+  } while (0)
+#endif
+
+namespace cudf {
+namespace test {
+
+/**---------------------------------------------------------------------------*
+ * @brief Wrapper for a gdf_column used for unit testing.
+ *
+ * An abstraction on top of a gdf_column that provides functionality for
+ * allocating, initializing, and otherwise managing gdf_columns for passing to
+ * libcudf APIs in unit testing.
+ *
+ * @tparam ColumnType The underlying data type of the column
+ *---------------------------------------------------------------------------**/
+template <typename ColumnType>
+struct column_wrapper {
+  /**---------------------------------------------------------------------------*
+   * @brief Copy constructor copies from another column_wrapper of the same
+   * type.
+   *
+   * @param other The column_wraper to copy
+   *---------------------------------------------------------------------------**/
+  column_wrapper(column_wrapper<ColumnType> const& other)
+      : data{other.data}, bitmask{other.bitmask}, the_column{other.the_column} {
+    the_column.data = data.data().get();
+    the_column.valid = bitmask.data().get();
+  }
+
+  // TODO Implement this via copy & swap. Need a swap function.
+  column_wrapper& operator=(column_wrapper<ColumnType> other) = delete;
+
+  ~column_wrapper() = default;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Implicit conversion operator to a gdf_column pointer.
+   *
+   * Allows for implicit conversion of a column_wrapper to a pointer to its
+   * underlying gdf_column.
+   *
+   * In this way, a column_wrapper can be passed directly into a libcudf API
+   * and will be implicitly converted to a pointer to its underlying gdf_column
+   * without the need to use the `get()` member.
+   *
+   * @return gdf_column* Pointer to the underlying gdf_column
+   *---------------------------------------------------------------------------**/
+  operator gdf_column*() { return &the_column; };
```

Comment:
I don't like this. Try `operator gdf_column&()` instead, and let the caller take the address. Also, aren't you worried about mutable access here?

Reply:
> Try `operator gdf_column&()` instead, and let the caller take the address.

I don't see how this is any better.

> Also, aren't you worried about mutable access here?

I'm not sure what you mean. 

This is simply a convenience such that if someone is calling an API that accepts a `gdf_column*` that they don't need to call `column_wrapper::get()`.